processing/src/test/java/org/apache/druid/java/util/emitter/core/ParametrizedUriEmitterTest.java

```diff
@@ -174,15 +178,22 @@ public void testEmitterWithParametrizedUriExtractor() throws Exception
           protected ListenableFuture<Response> go(Request request) throws JsonProcessingException
           {
             Assert.assertEquals("http://example.com/val1/val2", request.getUrl());
-            Assert.assertEquals(
-                StringUtils.format(
-                    "[%s,%s]\n",
-                    JSON_MAPPER.writeValueAsString(events.get(0)),
-                    JSON_MAPPER.writeValueAsString(events.get(1))
-                ),
-                StandardCharsets.UTF_8.decode(request.getByteBufferData().slice()).toString()
-            );
-
+            String event1Json = sortedJsonStrings(JSON_MAPPER.writeValueAsString(events.get(0)));
+            String event2Json = sortedJsonStrings(JSON_MAPPER.writeValueAsString(events.get(1)));
+            String expectedJsonContent = "[" + event1Json + "," + event2Json + "]";
+            byte[] requestBytes = request.getByteBufferData().slice().array();
+            String actualJsonContent = StandardCharsets.UTF_8.decode(ByteBuffer.wrap(requestBytes)).toString();
+            List<Map<String, Object>> jsonArray = new ObjectMapper().readValue(actualJsonContent, List.class);
+            List<Map<String, Object>> sortedJsonArray = new ArrayList<>();
+            for (Map<String, Object> map : jsonArray) {
+              Map<String, Object> sortedMap = new TreeMap<>(map);
+              sortedJsonArray.add(sortedMap);
+            }
+            ObjectWriter writer = new ObjectMapper().writerWithDefaultPrettyPrinter();
+            String sorted = writer.writeValueAsString(sortedJsonArray);
+            String expectedJsonNoSpaces = removeWhitespace(expectedJsonContent);
+            String actualJsonNoSpaces = removeWhitespace(sorted);
+            Assert.assertEquals(sorted, expectedJsonNoSpaces, actualJsonNoSpaces);
```

Comment:
It seems like you're removing whitespaces from sorted, but then checking if sorted is equal to expectedJsonNoSpaces, actualJsonNoSpaces. 
Please explain this line.